_posts/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins.md

```diff
@@ -0,0 +1,341 @@
+```
+yarn start --run-examples
+```
+
+Once Dashboards is up and running, navigate to the page and click on the menu on the left. From there, open the **developer examples** under OpenSearch Dashboards. Then click on **Multiple Data Sources Integration**. On the right side of the page, you'll find a comprehensive list of pickers, each accompanied by a brief introduction and a set of properties, listed below. Feel free to experiment with the example plugin. 
+
+<img src="/assets/media/blog-images/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins/multi_selectable.png">
+
+### 2.2 The data source as a saved object
+
+The data source is persisted as a saved object in the Dashboards system index. At the time of creation, a UUID is assigned as its unique identifier and data source metadata is also gathered, including the remote cluster version and associated plugins. This automatically generated data, as well as user-provided information including the title, endpoint, and authentication methods, are stored as attributes of the data source object. For the current schema of the object, refer to the [related code](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source/common/data_sources/types.ts#L8). 
+
+## 3 FAQ 
+
+Q: Is any UI component provided by `dataSourceManagemet` used in core OpenSearch?
+A: A data source selector is now used in Dev Tools, the tutorial, and on the saved object management page. You can see the component after setting `data_source.enabled` to true in `opensearch_dashbards.yml`.
+
+Q: Given that a dashboard is a saved object itself, will it not have a data connection associated with it?
+A: A data source, represented as a saved object, is identified by its unique saved object ID. Saved objects created from data sources have a references section containing the ID, name, and type. This reference helps you to identify the associated data source for the saved object. Here is an example of a index pattern reference:
+```
+"references": [
+    {
+        "id": "9dc4e190-08a0-11ef-901b-bb0301313422",
+        "type": "data-source",
+        "name": "dataSource"
+    }
+]
+```
+The visualization reference is slightly different because it points to the related index pattern and uses the index pattern ID to find the corresponding data source: 
+```
+"references": [
+    {
+        "name": "kibanaSavedObjectMeta.searchSourceJSON.index",
+        "type": "index-pattern",
+        "id": "f1613e10-08a6-11ef-901b-bb0301313422"
+        }
+]
+```
+
+
+## 4 List of supported plugins/features
+
+Starting in version 2.14, multiple data sources are supported in nine external Dashboards plugins: 
+
+* Index Management
+* Machine Learning
+* Search Relevance
+* Anomaly Detection
+* Maps
+* Security
+* Notification
+* Query Workbench
+* Trace Analytics 
+
+Additionally, we've extended this functionality to two core visualizations: Time Series Visual Builder (TSVB) and Region Map.
```

Comment:
```suggestion
Multiple data sources functionality is also now available with Time Series Visual Builder (TSVB) and Region Map visualization types.
```

Reply:
Should we also include Vega? See PR https://github.com/opensearch-project/project-website/pull/2837. FYI--user documentation is in process for this topic.

Reply:
Got it